File: backend/components/metiers/threed/tests_centralises/samples/validators/input/test_input_validators.js
// test_sample_input.js
// Tests avancés pour sample_input.js (validation d'input)
const validateInput = require('../../../../samples/validators/input/sample_input');

describe('validateInput', () => {
  it('valide une chaîne non vide', () => {
    expect(validateInput('valide')).toBe(true);
  });
  it('rejette une chaîne vide', () => {
    expect(validateInput('')).toBe(false);
  });
  it('rejette un type non string', () => {
    expect(validateInput(123)).toBe(false);
  });
});

File: backend/components/metiers/threed/tests_centralises/samples/validators/input/test_input_validators.py
# test___init__.py
"""
Tests avancés pour sample_input.py (validation d'input).
"""
from threed.samples.validators.input import sample_input


def test_validate_input_valid():
    assert sample_input.validate_input("valide") is True


def test_validate_input_empty():
    assert sample_input.validate_input("") is False


def test_validate_input_non_string():
    assert sample_input.validate_input(123) is False
